tests/integration/session/test_login.py

```diff
@@ -91,3 +110,41 @@ def test_login_with_missing_mandatory_claims_should_be_forbidden(self):
 
         # Then
         self.assertStatusForbidden()
+
+    def test_login_token_with_survey_url_should_redirect_to_survey(self):
+
+        survey_url = "http://eq-survey-register/my-test-schema"
+
+        # Given
+        token = self.token_generator.create_token_with_survey_url('textarea', 'test', survey_url)
+
+        # When
+        with HTTMock(self.register_mock):
+            self.get('/session?token=' + token)
+
+    def test_login_token_with_incorrect_survey_url_results_in_404(self):
+
+        survey_url = "http://eq-survey-register/my-test-schema"
```

Comment:
Although not required, it would be clearer if this url was different to the other one, something like `/my-test-schema-not-found`